qiskit/circuit/equivalence.py

```diff
@@ -284,7 +285,7 @@ def _raise_if_shape_mismatch(gate, circuit):
 
 def _rebind_equiv(equiv, query_params):
     equiv_params, equiv_circuit = equiv
-    param_map = dict(zip(equiv_params, query_params))
-    equiv = equiv_circuit.assign_parameters(param_map, inplace=False)
+    param_map = dict((x, y) for x, y in zip(equiv_params, query_params) if isinstance(x, Parameter))
+    equiv = equiv_circuit.assign_parameters(param_map, inplace=False, flat_input=True)
```

Comment:
This was a logical bug in the `EquivalenceLibrary` code that previously was being masked by a side-effect of `QuantumCircuit._unroll_params_dict` - it didn't raise errors on bad parameters in the map, it just silently dropped them.  For equivalences that have a mixture of symbolic and concrete parameters (there's at least one explicitly tested), this resulted in passing a map that looked like `{Parameter: float, float: float}`, which should be a typing error - a `float` isn't a valid `Parameter` and can't be bound.